include/ebpf_structs.h

```diff
@@ -83,20 +83,44 @@ typedef enum
 // Cross-platform BPF program types.
 enum bpf_prog_type
 {
-    BPF_PROG_TYPE_UNKNOWN,
+    BPF_PROG_TYPE_UNSPEC,
     BPF_PROG_TYPE_XDP,
     BPF_PROG_TYPE_BIND, // TODO(#333): replace with cross-platform program type
 };
 
+enum bpf_link_type
+{
+    BPF_LINK_TYPE_UNSPEC,
+    BPF_LINK_TYPE_PLAIN,
```

Comment:
link_type is which union element to use inside the bpf_link_type struct.   On Linux, attach type is in some union elements and not others.   Since we only have one union element in bpf_link_type so far, we only need one non-zero value.   If we start adding other things like ifindex then we can add other union elements (and corresponding link type values) at that time.